bookkeeper-server/src/main/java/org/apache/bookkeeper/client/AsyncCallback.java

```diff
@@ -48,6 +48,24 @@
          *          context object
          */
         void addComplete(int rc, LedgerHandle lh, long entryId, Object ctx);
```

Comment:
What was saying in the meeting,but wasn't very clear, is that if you add a default to addComplete, then all users of asyncAddEntry can take advantage of this latency reporting (not just the LedgerHandleAdv users).

As it is now, a user would have to override addComplete and addCompleteAdv to use the a latency stuff. If addComplete has a default, then existing code will continue to work, but a user who wants latency information can override only addCompleteAdv, ignoring addComplete.

The addComplete default can be empty, as the bookkeeper code only calls addCompleteAdv. addComplete is _only_ ever called by addCompleteAdv, in the case that addCompleteAdv isn't implemented.

Reply:
Yeah, that's better.

Reply:
@ivankelly Ah, right.  Doing that means that there is no longer an abstract method, so it no longer works as a FunctionalInterface, so lambda expressions don't work.  That would break some existing users.

Reply:
@ivankelly I wasn't able to find a good way to do this that didn't break existing users without this particular quirk that you have to implement both.

Reply:
Oops, meant to reply yesterday. Yes, since it would break the functional stuff, we have to add a new interface. In that case, I'd keep both callbacks completely separate, so that both can be functional.